src/System.Net.Mail/tests/Functional/AttachmentTest.cs

```diff
@@ -0,0 +1,98 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// See the LICENSE file in the project root for more information.
+//
+// AttachmentTest.cs - NUnit Test Cases for System.Net.MailAddress.Attachment
+//
+// Authors:
+//   John Luke (john.luke@gmail.com)
+//
+// (C) 2005 John Luke
+//
+
+using System.IO;
+using System.Text;
+using System.Net.Mime;
+using Xunit;
+
+namespace System.Net.Mail.Tests
+{
+    public class AttachmentTest
+    {
+        Attachment attach;
+
+        public AttachmentTest()
+        {
+            attach = Attachment.CreateAttachmentFromString("test", "attachment-name");
+        }
+
+        [Fact]
+        public void TestNullStream()
+        {
+            Stream s = null;
+            Assert.Throws<ArgumentNullException>(() => new Attachment(s, "application/octet-stream"));
+        }
+
+        [Fact]
+        public void ConstructorNullName()
+        {
+            Attachment attach = new Attachment(new MemoryStream(), null, "application/octet-stream");
+            Assert.Equal(null, attach.Name);
+        }
+
+        [Fact]
+        public void CreateAttachmentFromStringNullName()
+        {
+            Attachment.CreateAttachmentFromString("", null, Encoding.ASCII, "application/octet-stream");
```

Comment:
Same question about verifying lack of exception, and whether we should assert anything about the returned instance.  I'll stop asking about these.